python/pylibcudf/pylibcudf/io/text.pyx

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from cython.operator cimport dereference
+from libc.stdint cimport uint64_t
+from libcpp.memory cimport unique_ptr
+from libcpp.string cimport string
+from libcpp.utility cimport move
+
+from pylibcudf.column cimport Column
+from pylibcudf.libcudf.column.column cimport column
+from pylibcudf.libcudf.io cimport text as cpp_text
+
+cdef class ParseOptions:
+    """Parsing options for `multibyte_split`"""
+    def __init__(
+        self,
+        *,
+        byte_range=None,
+        strip_delimiters=False,
+    ):
+        self.c_options = cpp_text.parse_options()
+        if byte_range is not None:
+            c_byte_range_offset = byte_range[0]
+            c_byte_range_size = byte_range[1]
+            self.c_options.byte_range = cpp_text.byte_range_info(
+                c_byte_range_offset,
+                c_byte_range_size
+            )
+        self.c_options.strip_delimiters = strip_delimiters
+
+
+cdef class DataChunkSource:
+    """Data source for `multibyte_split`"""
+
+    def __init__(self):
+        raise ValueError(
+            "This class cannot be instantiated directly. "
+            "Use one of the make_source functions instead."
+        )
+
+    @staticmethod
+    cdef DataChunkSource from_source(unique_ptr[data_chunk_source] source):
+        cdef DataChunkSource datasource = DataChunkSource.__new__(DataChunkSource)
+        datasource.c_data_chunk_source = move(source)
+        return datasource
+
+
+cpdef DataChunkSource make_source(str data):
+    """
+    Creates a data source capable of producing device-buffered views
+    of the given string.
+
+    Parameters
+    ----------
+    data : str
+        The host data to be exposed as a data chunk source.
+
+    Returns
+    -------
+    DataChunkSource
+        The data chunk source for the provided host data.
+    """
+    cdef unique_ptr[data_chunk_source] c_source
+    cdef string c_data = data.encode()
+
+    with nogil:
+        c_source = cpp_text.make_source(c_data)
```

Comment:
My C++ is failing me here. The tests are failing with string data because I think libcudf requires the data to be alive for the lifetime of the source

From `data_source_factories.hpp`

> the host data to be exposed as a data chunk source. Its lifetime must be at least as long as the lifetime of the returned data_chunk_source.

What would be the best way to do that?

Reply:
I tried saving a reference to `c_data` on `DataChunkSource` but that didn't seem to work

Reply:
`c_data` lives on the stack, so it will be destructed when this function exits.

Try storing a `unique_ptr[string]` in the `DataChunkSource`?

Reply:
OK I got it.

I just stored in the input `string` initially `DataChunkSource` and passed that to the various `make_source*` methods